### PR TITLE
dotCMS/core#26900 Allow pass field variables to Monaco's Binary Field And JSON Field

### LINKED ISSUE #26900: Edit Content: Allow Configuration of Field Variables for Monaco Editor Options on Binary and JSON Fields
### User Story
As a dotCMS user, I want to be able to configure field variables that will pass options to the Monaco editor for Binary and JSON fields, so I can customize the appearance and behavior of the editor.

### Acceptance Criteria
- Allow users to configure field variables as an object in order to accept options for the Monaco editor for both Binary and JSON fields.

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- Users must be familiar with using the Monaco editor in order to configure the field variables.

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
N/A

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Allow pass field variables to Monaco's Binary Field 
* Allow pass field variables to Monaco's JSON Field

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
**JSON Field**
https://github.com/dotCMS/core/assets/1909643/7616ae84-130b-4cda-a0f4-de2a10e5387c

**Binary Field**
https://github.com/dotCMS/core/assets/1909643/17381d56-3873-4e3f-81be-be33de506ea2

### CODE REVIEW

File: core-web/libs/data-access/src/lib/dot-temp-file-upload/dot-temp-file-upload.service.ts [hunk 1]
```diff
@@ -7,17 +7,7 @@ import { catchError, map, pluck, take } from 'rxjs/operators';
 
 import { DotHttpErrorManagerService } from '@dotcms/data-access';
 import { CoreWebService } from '@dotcms/dotcms-js';
-
-export interface DotCMSTempFile {
```
Comment (oidacra) [inline core-web/libs/data-access/src/lib/dot-temp-file-upload/dot-temp-file-upload.service.ts:11]: Already exist a Interface with the exact same name. @rjvelazco in the other added metadata property.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts [hunk 2]
```diff
@@ -99,15 +112,33 @@ export class DotEditContentBinaryFieldComponent
...
+        private readonly dotMessageService: DotMessageService,
+        private readonly dotBinaryFieldEditImageService: DotBinaryFieldEditImageService,
+        private readonly DotBinaryFieldValidatorService: DotBinaryFieldValidatorService,
+        private readonly cd: ChangeDetectorRef
+    ) {
+        this.dotMessageService.init();
+    }
+
+    @Input({ required: true })
+    set field(contentTypeField: DotCMSContentTypeField) {
```
Comment (rjvelazco) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts:136]: Shouldn't `@Input()` be before the `constructor`?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts [hunk 3]
```diff
@@ -73,28 +62,39 @@ export class DotBinaryFieldEditorComponent implements OnInit {
     @Input() fileName = '';
     @Input() fileContent = '';
 
+    languageType = signal(DEFAULT_FILE_TYPE);
```
Comment (rjvelazco) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts:65]:
```suggestion
    private readonly languageType = signal(DEFAULT_FILE_TYPE);
```

**Review by rjvelazco — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts (hunk 2 above)
Comment (oidacra) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/dot-edit-content-binary-field.component.ts:136]: is by the 'set' the link order that in that order no idea.

**Review by hmoreras — APPROVED** [review]

File: core-web/apps/dotcms-binary-field-builder/project.json [hunk 4]
```diff
@@ -41,6 +41,11 @@
                 ]
             },
             "configurations": {
+                "localhost": {
```
Comment (fmontes) [inline core-web/apps/dotcms-binary-field-builder/project.json:44]: I don't think this is a good name for a configuration, shouldn't be like `development`

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.spec.ts [hunk 5]
```diff
@@ -126,17 +127,25 @@ describe('DotBinaryFieldEditorComponent', () => {
...
+
+            spectator.detectChanges();
+
             component.form.setValue({
                 name: 'script.js',
                 content: 'test'
             });
 
-            tick(1000);
+            component.form.updateValueAndValidity();
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.spec.ts:142]: Doesn't this get's trigger automatically when the `setValue` get's called?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts [hunk 6]
```diff
@@ -73,28 +62,39 @@ export class DotBinaryFieldEditorComponent implements OnInit {
...
+            ...DEFAULT_BINARY_FIELD_MONACO_CONFIG,
+            ...this._customMonacoOptions(),
+            language: this.languageType()
+        };
     });
 
-    editorOptions = EDITOR_CONFIG;
-    mimeType = '';
+    @Input()
+    set customMonacoOptions(customMonacoOptions: MonacoEditorConstructionOptions) {
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts:95]:
```suggestion
    set userMonacoOptions(customMonacoOptions: MonacoEditorConstructionOptions) {
```

File: core-web/libs/edit-content/src/lib/utils/functions.util.ts [hunk 7]
```diff
@@ -113,3 +114,59 @@ export const transformLayoutToTabs = (
...
+ * @param {string} value - The value to be checked
+ * @returns {boolean} - True if the value is a valid JSON string, false otherwise
+ */
+export const isValidJson = (value: string): boolean => {
+    try {
+        const json = JSON.parse(value);
+
+        return json !== null && typeof json === 'object' && !Array.isArray(json);
+    } catch (e) {
+        console.warn(`${value} is not a valid JSON`);
```
Comment (fmontes) [inline core-web/libs/edit-content/src/lib/utils/functions.util.ts:130]: If the user look and the console and this is the only error they see, they will not know what to fix, I think we can do better here, specially for this `fieldVariable` options

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

File: core-web/apps/dotcms-binary-field-builder/project.json (hunk 4 above)
Comment (oidacra) [reply core-web/apps/dotcms-binary-field-builder/project.json:44]: Deleted, used development instead.

File: core-web/libs/edit-content/src/lib/utils/functions.util.ts (hunk 7 above)
Comment (oidacra) [reply core-web/libs/edit-content/src/lib/utils/functions.util.ts:130]: It return a boolean. That can be handled direct in the origin I think.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts (hunk 6 above)
Comment (oidacra) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.ts:95]: Renamed.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.spec.ts (hunk 5 above)
Comment (oidacra) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-binary-field/components/dot-binary-field-editor/dot-binary-field-editor.component.spec.ts:142]: Yes. It does. Removed, Only is necessary if we manipulate the validators.

**Pull request merged**